crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1893,19 +1909,83 @@ where
                 ctx: ExprContext::Store,
                 range: _,
             }) => {
-                if let Some(unpack) = self
-                    .current_assignment()
-                    .as_ref()
-                    .and_then(CurrentAssignment::unpack)
-                {
-                    self.register_attribute_assignment(
-                        object,
-                        attr,
-                        AttributeAssignment::Unpack {
-                            attribute_expression_id: expression_id,
-                            unpack,
-                        },
-                    );
+                match self.current_assignment() {
+                    Some(CurrentAssignment::Assign { node, unpack, .. }) => {
+                        // SAFETY: `value` and `expr` belong to the `self.module` tree
+                        #[allow(unsafe_code)]
+                        let assignment = AssignmentDefinitionKind::new(
+                            TargetKind::from(unpack),
```

Comment:
In this PR, `TargetKind` still has only two variants, `Name` and `Sequence`, but now we end up using `Name` also when the target is an attribute. This should be fixed in some way, since it's misleading.

It seems maybe we don't ever need to distinguish between name target and attribute target using the `TargetKind`? In which case we could rename the `Name` variant to `NameOrAttribute`. Or we could add a third variant, `Attribute`, and make sure we pass `Attribute` variant here. The latter seems nicer in that we preserve more information, but if we don't need that information maybe it's not worth it.